core/src/banking_stage/packet_deserializer.rs

```diff
@@ -106,29 +106,35 @@ impl PacketDeserializer {
         ) -> Result<ImmutableDeserializedPacket, PacketFilterFailure>,
     ) -> ReceivePacketResults {
         let mut packet_stats = PacketReceiverStats::default();
-        let mut deserialized_packets = Vec::with_capacity(packet_count);
-
-        for banking_batch in banking_batches {
-            for packet_batch in banking_batch.iter() {
-                let packet_indexes = Self::generate_packet_indexes(packet_batch);
-
-                saturating_add_assign!(
-                    packet_stats.passed_sigverify_count,
-                    packet_indexes.len() as u64
-                );
-                saturating_add_assign!(
-                    packet_stats.failed_sigverify_count,
-                    packet_batch.len().saturating_sub(packet_indexes.len()) as u64
-                );
-
-                deserialized_packets.extend(Self::deserialize_packets(
-                    packet_batch,
-                    &packet_indexes,
-                    &mut packet_stats,
-                    &packet_filter,
-                ));
-            }
-        }
+        let mut errors = 0_usize;
+        let deserialized_packets: Vec<_> = banking_batches
+            .iter()
+            .flat_map(|banking_batch| banking_batch.iter())
+            .flat_map(|batch| batch.iter())
+            .filter(|pkt| !pkt.meta().discard())
+            .filter_map(|pkt| {
+                match ImmutableDeserializedPacket::new(pkt)
+                    .and_then(|pkt| packet_filter(pkt).map_err(Into::into))
+                {
+                    Ok(pkt) => Some(pkt),
+                    Err(err) => {
+                        saturating_add_assign!(errors, 1);
+                        packet_stats.increment_error_count(&err);
+                        None
+                    }
+                }
+            })
+            .collect();
+        saturating_add_assign!(
+            packet_stats.passed_sigverify_count,
+            deserialized_packets.len().saturating_add(errors) as u64
+        );
+        saturating_add_assign!(
+            packet_stats.failed_sigverify_count,
+            packet_count
+                .saturating_sub(deserialized_packets.len())
+                .saturating_sub(errors) as u64
+        );
```

Comment:
So it means:
```rust
packet_stats.failed_sigverify_count +=
    (packet_count - deserialized_packets.len() - errors) as u64;
```

which means that we assume that there are only two types of error -- sigverify and deserialize.

Reply:
Exactly.